### PR TITLE
Fix/18884 migrate avatar network

### LINKED ISSUE #18884: Migrate components to TS: AvatarNetwork
### Description
Migrate `AvatarNetwork` component to TypeScript

### Technical Details
- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` follow [TS conventions](https://www.notion.so/Use-TypeScript-9ec7bd8fc8cf4d82b81cd8943c4707cd?pvs=4#7e6410b21100455a9a08442108f19d78)
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation 

### Acceptance Criteria
- Component has been converted to TypeScript

### PR DESCRIPTION
**Pull request by thebinij**
## Explanation

* Closes #18884 

### Before

https://github.com/MetaMask/metamask-extension/assets/93417120/e484ba70-5094-4ff8-9f31-03d0ec8ad068

### After

https://github.com/MetaMask/metamask-extension/assets/93417120/7a838974-03af-49ae-b413-d00aec3ee7ec

## Manual Testing Steps

No functional changes

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by thebinij** [comment]
Hi @georgewrmarshall, I have updated the code after you suggestions in the [Old PR](https://github.com/MetaMask/metamask-extension/pull/19048). Please review it. I need your help to understand causation of error in test-deps-depcheck. Thanks

File: ui/components/component-library/avatar-network/avatar-network.types.ts
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-network/avatar-network.types.ts:20]: I think we could extend `AvatarBaseProps` here instead of BoxProps and remove `backgroundColor`, `borderColor`,  `color`, `as` types altogether?

File: ui/components/component-library/avatar-base/avatar-base.types.ts
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-base/avatar-base.types.ts:24]: Not sure about this because if we carry on this pattern we will just end up with many variant types that AvatarBase accepts. There must be a TypeScript pattern that uses an argument type? I'm willing to proceed until we develop that pattern

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. Left one suggestion that I missed in my last review. And it will need a rebase

File: ui/components/component-library/avatar-network/avatar-network.types.ts
Comment (thebinij) [reply ui/components/component-library/avatar-network/avatar-network.types.ts]: Yeah, that would be great.

File: ui/components/component-library/avatar-network/avatar-network.types.ts
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-network/avatar-network.types.ts:18]: I think we should make this required so the AvatarNetwork always has a child

**Review by georgewrmarshall — COMMENTED** [review]
Hey @thebinij, one more suggestion

**Review by georgewrmarshall — COMMENTED** [review]
Hey @thebinij, code is looking good could you please add a screencast from storybook of the component docs and stories to the PR description. That way we can ensure there is no visual regression and everything works as expected. Thanks!

**Comment by thebinij** [comment]
> Hey @thebinij, code is looking good could you please add a screencast from storybook of the component docs and stories to the PR description. That way we can ensure there is no visual regression and everything works as expected. Thanks!

Hi @georgewrmarshall, I have included the before and after screencast to the description.

**Comment by georgewrmarshall** [comment]
Hey @thebinij, Thanks for your time and effort in creating this pull request. As you're probably aware we have identified an issue related to the typing of the `Box` component, specifically with the polymorphic `as` prop. To address this issue, we have an open ticket at https://github.com/MetaMask/metamask-extension/issues/19239 and a draft pull request at https://github.com/MetaMask/metamask-extension/pull/19363 that needs to be merged before we can proceed with your PR. Thanks for your patience

**Comment by thebinij** [comment]
Hi @georgewrmarshall, Is there anything I can update in the PR before so you approve to merge it?

**Comment by georgewrmarshall** [comment]
Blocked by https://github.com/MetaMask/metamask-extension/pull/19572

File: ui/components/component-library/avatar-network/README.mdx
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-network/README.mdx:47]: Nice

File: ui/components/component-library/avatar-network/avatar-network.types.ts
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-network/avatar-network.types.ts:10]: Unfortunately, we can't reassign enums we have to define them again
```suggestion
  Xs = 'xs',
  Sm = 'sm',
  Md = 'md',
  Lg = 'lg',
  Xl = 'xl',
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Unfortunately we can't reassign enums you can sort of see what happens when using the storybook controls

Screenshot 2023-07-21 at 11 05 04 AM


Also not sure why the props table isn't showing up?

Screenshot 2023-07-21 at 11 04 39 AM

It would be helpful to update the Before/After screencasts to ensure the storybook still works
https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-avatarnetwork--docs

File: ui/components/component-library/avatar-network/avatar-network.stories.tsx
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-network/avatar-network.stories.tsx:11]: Can we use the TS version of the `Box` from the component library?
```suggestion
import { Box } from '..'
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looking good! One suggestion to remove the JS version of the `Box` from `avatar-network.stories.tsx`

And can we update this line in the `ui/components/component-library/avatar-network/README.mdx` to point to the TS Box props
```
The `AvatarNetwork` accepts all props below as well as all [Box](/docs/components-ui-box--default-story#props) component props
```
to
```
The `AvatarNetwork` accepts all props below as well as all [Box](/docs/components-componentlibrary-box--docs#props) component props
```

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Thanks for your contribution @thebinij and supporting @garrettbear

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**
